### PR TITLE
LIMS-5220:Orders: Table with add: Allow user to add any number of the suborders records not only 5 suborders

### PR DESCRIPTION
**Pull request by omnia353**
Orders: Table with add: Allow user to add any number of the suborders records not only 5 suborders
https://modeso.atlassian.net/browse/LIMS-5220

```
C:\Users\omnia\Documents\1limsNewMaster\1lims-automation>nosetests -vs --logging-level=WARNING ui_testing/testcases/basic_tests/test006_orders.py -m test091 --tc-file=config.ini
c:\users\omnia\lib\site-packages\nose\plugins\manager.py:394: RuntimeWarning: Unable to load plugin noseprogressive = noseprogressive:ProgressivePlugin: No module named 'fcntl'
  warn("Unable to load plugin %s: %s" % (ep, e),
2020-09-07 14:52:58.364 | INFO     | api_testing.apis.base_api:_get_authorized_session:46 - Get authorized api session.
2020-09-07 14:52:58.368 | INFO     | api_testing.apis.base_api:_get_authorized_session:47 - admin:admin
2020-09-07 14:52:59.082 | INFO     | api_testing.apis.base_api:_get_authorized_session:58 - session ID : gTvXEkYa1RucLqDPHKxCR089FnrSgvk6tBAFzoo84MA .....

DevTools listening on ws://127.0.0.1:56665/devtools/browser/5a742359-6b0f-4b51-bdd4-24f8ce606d32
Orders: Table with add: Allow user to add any number of the suborders records not only 5 suborders ...
2020-09-07 14:53:44.385 | INFO     | ui_testing.testcases.base_test:setUp:27 - Test case : test091_add_any_number_of_suborders
2020-09-07 14:53:48.889 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-07 14:53:48.924 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:53:49.789 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:53:50.683 | INFO     | api_testing.apis.test_unit_api:set_name_configuration_name_only:427 - set test unit configuration
2020-09-07 14:53:51.745 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-07 14:53:51.761 | INFO     | api_testing.apis.orders_api:set_configuration:432 - set order configuration
2020-09-07 14:53:52.640 | INFO     | api_testing.apis.base_api:set_configuration:73 - status code: 1
2020-09-07 14:53:52.650 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/articles
2020-09-07 14:53:53.500 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:53.508 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/materialTypes
2020-09-07 14:53:53.683 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:53.695 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits
2020-09-07 14:53:53.886 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:53.895 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testUnits/get/874
2020-09-07 14:53:54.026 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:54.048 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/testPlans
2020-09-07 14:53:54.264 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:54.283 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders
2020-09-07 14:53:54.742 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:54.749 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=2&deleted=0
2020-09-07 14:53:54.913 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:54.918 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=3&deleted=0
2020-09-07 14:53:55.060 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:55.067 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=4&deleted=0
2020-09-07 14:53:55.300 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:55.304 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=5&deleted=0
2020-09-07 14:53:55.465 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:55.472 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=6&deleted=0
2020-09-07 14:53:55.631 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:55.648 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=7&deleted=0
2020-09-07 14:53:55.780 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:55.798 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=8&deleted=0
2020-09-07 14:53:56.057 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:53:56.065 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test091_add_any_number_of_suborders:3089 - edit order with No 78-2020
2020-09-07 14:53:57.874 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:582 - wait until page is loaded
2020-09-07 14:53:58.573 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:53:59.459 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:54:00.336 | INFO     | api_testing.apis.base_api:wrapper:122 - GET : https://automation.1lims.com/api/orders/suborder/?orderId=8&deleted=0
2020-09-07 14:54:00.490 | INFO     | api_testing.apis.base_api:wrapper:129 - Status code: 1
2020-09-07 14:54:00.497 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test091_add_any_number_of_suborders:3094 - create extra 2 suborders
2020-09-07 14:54:00.577 | INFO     | ui_testing.pages.order_page:create_new_suborder:370 - Add new suborder.
2020-09-07 14:54:01.079 | INFO     | ui_testing.pages.order_page:create_new_suborder:376 - Set material type : Raw Material
2020-09-07 14:54:08.820 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:54:10.321 | INFO     | ui_testing.pages.order_page:create_new_suborder:379 - Set article name : f520efbf98
2020-09-07 14:54:10.325 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:54:21.433 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:54:22.849 | INFO     | ui_testing.pages.order_page:create_new_suborder:386 - Set test plan : ee2c3d6fda
2020-09-07 14:54:30.917 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:54:37.434 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:54:38.463 | INFO     | ui_testing.pages.order_page:get_suborder_data:419 - getting main order data
2020-09-07 14:54:38.675 | INFO     | ui_testing.pages.order_page:get_suborder_data:426 - getting suborders data
2020-09-07 14:54:52.733 | INFO     | ui_testing.pages.order_page:create_new_suborder:370 - Add new suborder.
2020-09-07 14:54:53.574 | INFO     | ui_testing.pages.order_page:create_new_suborder:376 - Set material type : Raw Material
2020-09-07 14:55:01.840 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:03.152 | INFO     | ui_testing.pages.order_page:create_new_suborder:379 - Set article name : f520efbf98
2020-09-07 14:55:03.155 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:14.032 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:14.903 | INFO     | ui_testing.pages.order_page:create_new_suborder:386 - Set test plan : ee2c3d6fda
2020-09-07 14:55:23.152 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:29.837 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:30.904 | INFO     | ui_testing.pages.order_page:get_suborder_data:419 - getting main order data
2020-09-07 14:55:31.088 | INFO     | ui_testing.pages.order_page:get_suborder_data:426 - getting suborders data
2020-09-07 14:55:45.760 | INFO     | ui_testing.pages.base_pages:save:55 - save the changes
2020-09-07 14:55:45.763 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:46.976 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:49.813 | INFO     | ui_testing.testcases.basic_tests.test006_orders:test091_add_any_number_of_suborders:3103 - duplicate 5 suborders
2020-09-07 14:55:53.748 | INFO     | ui_testing.pages.base_pages:save:55 - save the changes
2020-09-07 14:55:53.752 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:55:55.123 | DEBUG    | ui_testing.pages.base_pages:sleep_tiny:39 - wait up to 0.5 sec
2020-09-07 14:56:05.238 | INFO     | ui_testing.testcases.base_test:tearDown:33 - go to dashboard page
2020-09-07 14:56:07.618 | INFO     | ui_testing.testcases.base_test:tearDown:35 - TearDown.     
ok

----------------------------------------------------------------------
Ran 1 test in 190.702s

OK

```

### CODE REVIEW

**Comment by omnia353** [comment]
i added 2 suborders instead of 10 after choosing order with multiple suborder (no_suborders greater than or equal 5)
as 10 suborders takes too much time and the concept is add suborder over the main 5 subordes

File: ui_testing/testcases/basic_tests/test006_orders.py [hunk 1]
```diff
@@ -3078,3 +3078,31 @@ def test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived(sel
             self.orders_page.filter_by_analysis_number(filter_text=analysis_no[i])
             self.assertEqual(len(self.order_page.result_table()) - 1, 0)
 
+    def test091_add_any_number_of_suborders(self):
+        '''
+        Orders: Table with add: Allow user to add any number of the suborders records not only 5 suborders
+        LIMS-5220
+        '''
+        testPlan = TestPlanAPI().create_completed_testplan_random_data()
+        order = self.orders_api.get_order_with_multiple_sub_orders(no_suborders=4)
```
Comment (ghada-37) [inline ui_testing/testcases/basic_tests/test006_orders.py:3087]: this test case will fail if no order with multiple suborders above 4 not found
@0xIslamTaha , so should I ask omnia to modify it.
or it's accepted ?
If accepted you can merge code

Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test006_orders.py:3087]: @ghada-37 @omnia353 
Create an order with 5 suborders using the API and use it here

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test006_orders.py:3087]: ok

**Comment by omnia353** [comment]
@ghada-37 i send payload with length 5 and i receive it with length 1

**Pull request closed without merging**